src/spotify_to_tidal/sync.py

```diff
@@ -192,6 +221,21 @@ def search(self, spotify_session: spotipy.Spotify, spotify_playlist):
                 results.append( (track, None) )
         return (results, cache_hits)
 
+    def _populate(self, spotify_track: t_spotify.SpotifyTrack, tidal_tracks: List[tidalapi.Track]) -> bool:
+        for idx, tidal_track in list(enumerate(tidal_tracks)):
+            if match(tidal_track, spotify_track):
+                self.data[spotify_track['id']] = tidal_track.id
+                tidal_tracks.pop(idx)
```

Comment:
After popping won't the index shift by 1 and you skip a track? You're probably better off doing the iterating and modification of the list in two separate loops, or building a new list of tidal tracks that didn't match and returning that.

Reply:
Oh my bad,  I see now that it returns on the next line, so popping is safe